src/Sylius/Bundle/CoreBundle/Validator/Constraints/TranslationForExistingLocalesValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Sylius\Component\Core\Model\ProductVariant;
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Sylius\Component\Product\Model\ProductVariantTranslationInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class TranslationForExistingLocalesValidator extends ConstraintValidator
+{
+    public function __construct(private RepositoryInterface $localeRepository)
+    {
+    }
+
+    public function validate($value, Constraint $constraint): void
+    {
+        Assert::isInstanceOf($value, ProductVariant::class);
```

Comment:
```suggestion
        Assert::isInstanceOf($value, TranslatableInterface::class);
```

No point in limiting it only to the `ProductVariant`

Reply:
Good catch